File: week2 - Arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>
int count_sentences(string text);
int count_words(string text);
int count_letters(string text);
int main(int argc, string argv[])
{
    int nl; // numero de letras
    int nw;
    int ns;
    float index;
    string text = get_string("Text: ");
    nl = count_letters(text);
    nw = count_words(text);
    ns = count_sentences(text);
    float l = nl * 100.0 / nw;
    float s = ns * 100.0 / nw;
    index = 0.0588 * l - 0.296 * s - 15.8;
    index = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }

    //printf("%i\n", nw);
    //printf("%i\n", nl);
    //printf("%i\n", ns);
}
int count_letters(string text)
{
    int n = 0;
    for (int j = 0, k = strlen(text); j < k; j++)
    {
        char l = text[j];
        if (isalpha(l))
        {
            n++;
        }
    }
    return n;
}
int count_words(string text)
{
    int n = 0;
    for (int j = 0, k = strlen(text); j < k; j++)
    {
        char l = text[j];
        if (l == 32)
        {
            n++;
        }
    }
    n++;
    return n;
}
int count_sentences(string text)
{
    int n = 0;
    for (int j = 0, k = strlen(text); j < k; j++)
    {
        char l = text[j];
        if (l == 33 || l == 63 || l == 46)
        {
            n++;
        }
    }
    return n;
}

File: week6 - Python/sentimental-readability/readability.py
# TODO
def main():
    text = input("Text: ")
    nl = count_letters(text)
    nw = count_words(text)
    ns = count_sentences(text)
    l = nl * 100.0 / nw
    s = ns * 100.0 / nw
    index = 0.0588 * l - 0.296 * s - 15.8
    index = round(index)
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade " + str(index))


def count_letters(text):
    n = 0
    j = 0
    k = len(text)
    while j < k:
        l = text[j]
        j += 1
        if (l.isalpha()) == True:
            n += 1
    return n


def count_words(text):
    n = 0
    j = 0
    k = len(text)
    while j < k:
        l = ord(text[j])
        j += 1
        if l == 32:
            n += 1
    n += 1
    return n


def count_sentences(text):
    n = 0
    j = 0
    k = len(text)
    while j < k:
        l = ord(text[j])
        j += 1
        if l == 33 or l == 63 or l == 46:
            n += 1

    return n


if __name__ == "__main__":
    main()